doc/adr/0012-modules-structure.md

```diff
@@ -0,0 +1,105 @@
+# 12. Modules' structure
+
+Date: 2022-12-28
+
+## Status
+
+Accepted
+
+## Context
+
+The existing structure of tessellation modules was mixed, hence hard to maintain.
+
+- `shared` contains both base classes and extensions, and tessellation schema,
+- `keytool` contains key-pair generator and exposes key-related operations,
+- `wallet` contains operations related to transactions,
```

Comment:
"related to transactions" or "related to creating transactions"? 🤔

Reply:
fixed :+1: